homeassistant/components/sensor/swiss_hydrological_data.py

```diff
@@ -12,137 +12,126 @@
 
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
-    TEMP_CELSIUS, CONF_NAME, STATE_UNKNOWN, ATTR_ATTRIBUTION)
+    CONF_NAME, STATE_UNKNOWN, ATTR_ATTRIBUTION)
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
 
-REQUIREMENTS = ['xmltodict==0.11.0']
+REQUIREMENTS = ['swisshydrodata==0.0.2']
 
 _LOGGER = logging.getLogger(__name__)
-_RESOURCE = 'http://www.hydrodata.ch/xml/SMS.xml'
 
 CONF_STATION = 'station'
+CONF_MEASUREMENTS = 'measurements'
 CONF_ATTRIBUTION = "Data provided by the Swiss Federal Office for the " \
                    "Environment FOEN"
 
-DEFAULT_NAME = 'Water temperature'
-
-ICON = 'mdi:cup-water'
-
-ATTR_LOCATION = 'location'
-ATTR_UPDATE = 'update'
-ATTR_DISCHARGE = 'discharge'
-ATTR_WATERLEVEL = 'level'
-ATTR_DISCHARGE_MEAN = 'discharge_mean'
-ATTR_WATERLEVEL_MEAN = 'level_mean'
-ATTR_TEMPERATURE_MEAN = 'temperature_mean'
-ATTR_DISCHARGE_MAX = 'discharge_max'
-ATTR_WATERLEVEL_MAX = 'level_max'
-ATTR_TEMPERATURE_MAX = 'temperature_max'
+DEFAULT_NAME = 'SwissHydroData'
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_STATION): vol.Coerce(int),
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_MEASUREMENTS): vol.Any([
+        'temperature',
+        'level',
+        'discharge',
+        'min_temperature',
+        'min_level',
+        'min_discharge',
+        'max_temperature',
+        'max_level',
+        'max_discharge',
+        'mean_temperature',
+        'mean_level',
+        'mean_discharge'
+    ])
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Swiss hydrological sensor."""
-    import xmltodict
-
     name = config.get(CONF_NAME)
     station = config.get(CONF_STATION)
+    measurements = config.get(CONF_MEASUREMENTS)
 
-    try:
-        response = requests.get(_RESOURCE, timeout=5)
-        if any(str(station) == location.get('@StrNr') for location in
-               xmltodict.parse(response.text)['AKT_Data']['MesPar']) is False:
-            _LOGGER.error("The given station does not exist: %s", station)
-            return False
-    except requests.exceptions.ConnectionError:
-        _LOGGER.error("The URL is not accessible")
+    data = HydrologicalData(station)
+
+    response = requests.get(
```

Comment:
Should be tested with `swisshydrodata` as `shd` should be created here anyway and passed to `HydrologicalData`.